packages/contracts-bedrock/src/deploy-config.ts

```diff
@@ -0,0 +1,339 @@
+
+  /**
+   * Chain ID for the L1 network.
+   */
+  l1ChainID: number
+
+  /**
+   * Chain ID for the L2 network.
+   */
+  l2ChainID: number
+
+  /**
+   * Number of seconds in between each L2 block.
+   */
+  l2BlockTime: number
+
+  /**
+   * Sequencer batches may not be more than maxSequencerDrift seconds after the L1 timestamp of the
+   * end of the sequencing window end.
+   */
+  maxSequencerDrift: number
+
+  /**
+   * Number of L1 blocks per sequencing window.
+   */
+  sequencerWindowSize: number
+
+  /**
+   * Number of seconds (w.r.t. L1 time) that a frame can be valid when included in L1.
+   */
+  channelTimeout: number
+
+  /**
+   * Address of the key the sequencer uses to sign blocks on the P2P layer.
+   */
+  p2pSequencerAddress: string
+
+  /**
+   * L1 address that batches are sent to.
+   */
+  batchInboxAddress: string
+
+  /**
+   * Acceptable batch-sender address, to filter transactions going into the batchInboxAddress on L1 for data.
+   * Warning: this address is hardcoded now, but is intended to become governed via L1.
+   */
+  batchSenderAddress: string
+
+  /**
+   * Output Oracle submission interval in L2 blocks.
+   */
+  l2OutputOracleSubmissionInterval: number
+
+  /**
+   * Starting block number for the output oracle.
+   */
+  l2OutputOracleStartingBlockNumber: number
+
+  /**
+   * Starting timestamp for the output oracle.
+   */
+  l2OutputOracleStartingTimestamp: number
+
+  /**
+   * Address of the L2 output oracle proposer.
+   */
+  l2OutputOracleProposer: string
+
+  /**
+   * Address of the L2 output oracle challenger.
+   */
+  l2OutputOracleChallenger: string
+
+  /**
+   * Output finalization period in seconds.
+   */
+  finalizationPeriodSeconds: number
+
+  /**
+   * Owner of the SystemConfig contract.
+   */
+  systemConfigOwner: string
+}
+
+/**
+ * Optional deployment configuration when spinning up an L1 network as part of the deployment.
+ */
+interface OptionalL1DeployConfig {
+  l1BlockTime: number
+  l1GenesisBlockNonce: string
+  cliqueSignerAddress: string
+  l1GenesisBlockGasLimit: string
+  l1GenesisBlockDifficulty: string
+  l1GenesisBlockMixHash: string
+  l1GenesisBlockCoinbase: string
+  l1GenesisBlockNumber: string
+  l1GenesisBlockGasUsed: string
+  l1GenesisBlockParentHash: string
+  l1GenesisBlockBaseFeePerGas: string
+}
+
+/**
+ * Optional deployment configuration when spinning up an L2 network as part of the deployment.
+ */
+interface OptionalL2DeployConfig {
+  l2GenesisBlockNonce: string
+  l2GenesisBlockExtraData: string
+  l2GenesisBlockGasLimit: string
+  l2GenesisBlockDifficulty: string
+  l2GenesisBlockMixHash: string
+  l2GenesisBlockCoinbase: string
+  l2GenesisBlockNumber: string
+  l2GenesisBlockGasUsed: string
+  l2GenesisBlockParentHash: string
+  l2GenesisBlockBaseFeePerGas: string
+  optimismBaseFeeRecipient: string
+  optimismL1FeeRecipient: string
+  l2CrossDomainMessengerOwner: string
+  gasPriceOracleOverhead: number
+  gasPriceOracleScalar: number
```

Comment:
Some of these values are technically optional but i'm not sure if they should be as it would degrade usage of the system

Reply:
I can make them required in another PR